app/Controller/AppController.php

```diff
@@ -406,6 +406,12 @@ public function beforeFilter()
                     if (!$this->_isRest()) {
                         $this->redirect(array('controller' => 'users', 'action' => 'change_pw', 'admin' => false));
                     }
+                } elseif (Configure::read('Security.email_otp_enabled') && !$this->_isRest() && !in_array($this->request->here, array($base_dir.'/users/terms', $base_dir.'/users/email_otp', $base_dir.'/users/change_pw', $base_dir.'/users/logout', $base_dir.'/users/login'))) {
+                    $redis = $this->{$this->modelClass}->setupRedis();
+                    $otp_authed = $redis->get('misp:otp_authed:'.$this->Auth->user('id'));
```

Comment:
token should not be on user_id , this would be a vulnerability in the use-case:
- legit user is logged in with OTP-is already validated.
- malicious user logs in with stolen credentials, (s)he does not get the request to the OTP.